src/Lykke.Service.Balances.Services/Wallet/CachedWalletsRepository.cs

```diff
@@ -23,32 +25,59 @@ public class CachedWalletsRepository : ICachedWalletsRepository
             [NotNull] IDistributedCache cache,
             [NotNull] IWalletsRepository repository,
             TimeSpan cacheExpiration,
-            [NotNull] ILogFactory logFactory)
+            [NotNull] ILogFactory logFactory,
+            [NotNull] IDatabase redisDatabase,
+            [NotNull] string partitionKey)
         {
-            _cache = cache ?? throw new ArgumentNullException(nameof(cache));
             _repository = repository ?? throw new ArgumentNullException(nameof(repository));
             _cacheExpiration = cacheExpiration;
+            _redisDatabase = redisDatabase ?? throw new ArgumentNullException(nameof(redisDatabase));
+            _partitionKey = partitionKey ?? throw new ArgumentNullException(nameof(partitionKey));
             _log = logFactory.CreateLog(this);
         }
 
         public async Task<IReadOnlyList<IWallet>> GetAllAsync(string walletId)
         {
-            return await _cache.TryGetAsync(
-                GetAllBalancesCacheKey(walletId),
-                async () => (await _repository.GetAsync(walletId))
-                    .Select(CachedWalletModel.Create)
-                    .ToArray(),
-                slidingExpiration: _cacheExpiration,
-                log: _log);
+            // todo: refactor most of code below into RedisCacheExtensions method 'TryHashGetAllAsync'
+            try
+            {
+                var balances = await _redisDatabase.HashGetAllAsync(GetCacheKey(walletId));
+                if (balances != null && balances.Length > 0)
+                {
+                    return balances.Select(x => CacheSerializer.Deserialize<CachedWalletModel>(x.Value)).ToList();
+                }
+            }
+            catch (RedisConnectionException ex)
+            {
+                _log.Warning("Redis cache is not available", ex);
+            }
+
+            var result = (await _repository.GetAsync(walletId)).Select(CachedWalletModel.Create).ToArray();
+
+            try
+            {
+                await _redisDatabase.HashSetAsync(
```

Comment:
We have an `ILog` private instance here but do not pass it into Redis extension method. May be, we actually shoud, to have some more details about possible failure.